app/modules/sightings/resources.py

```diff
@@ -57,21 +80,182 @@ def get(self, args):
+        i = 0
+        while i < len(data['encounters']):
+            enc_data = data['encounters'][i]
+            # TODO handle regular .assets flavor
+            # TODO genericize this across here and encounters
+            if 'assetReferences' in enc_data and isinstance(
+                enc_data['assetReferences'], list
+            ):
+                enc_data['_paths_wanted'] = []
+                for aref in enc_data['assetReferences']:
+                    if (
+                        not isinstance(aref, dict)
+                        or 'path' not in aref
+                        or 'transactionId' not in aref
+                    ):
+                        log.error(
+                            'Sighting.post malformed assetReferences data: %r' % (aref)
+                        )
+                        _cleanup_post_and_abort(
+                            None, None, 'Malformed assetReferences data'
+                        )
+                    enc_data['_paths_wanted'].append(aref['path'])
+                    key = ':'.join((aref['transactionId'], aref['path']))
+                    if key in arefs_found:
+                        arefs_found[key]['encs'].append(i)
+                    else:
+                        arefs_found[key] = aref.copy()
+                        arefs_found[key]['encs'] = [i]
+            i += 1
+
+        # now we have a mapping of encounters to assetReferences... so we try to create the submission + assets
+        #   first we make sure we have the files we need so we can abort if not (they may fail later for other reasons)
+        from app.extensions.tus import tus_upload_dir
+
+        transaction_id = None  # see note above about single common transaction id
+        for key in arefs_found:
+            aref = arefs_found[key]
+            transaction_id = aref['transactionId']
+            file_path = os.path.join(
+                tus_upload_dir(current_app, transaction_id=transaction_id),
+                aref['path'],
+            )
+            try:
+                sz = os.path.getsize(file_path)  # 2for1
+            except OSError as err:
+                log.error(
+                    'Sighting.post OSError %r assetReferences data: %r' % (err, aref)
+                )
+                _cleanup_post_and_abort(None, None, 'File not found: ' + str(aref))
+            if sz < 1:
+                log.error(
+                    'Sighting.post zero-size file for assetReferences data: %r' % (aref)
+                )
+                _cleanup_post_and_abort(None, None, 'File empty: ' + str(aref))
+
+        # files seem to exist in uploads dir, so lets move on
+        from app.modules.submissions.models import Submission
+
+        pub = False
+        owner_guid = None
+        if current_user is not None and not current_user.is_anonymous:
+            owner_guid = current_user.guid
+            pub = True
+        submission, assets_added = Submission.create_submission_from_tus(
```

Comment:
I'm really not seeing the benefit of assets_added, Surely it's the same as submission.assets?

Reply:
this is kind of leftover from talking with karen and ben both about potential use cases where users upload *more images than they ultimately want to use* to the transaction tmp dir -- and then needing a way to use (import as assets) only a subset of those.    i just havent finished this code yet.. but basically the code here where it builds up `ahref` to reference all the assets in all the encounters in this sighting has that info in it.  it will balk if the tmp dir is _missing_ something, but makes no attempt to cry foul if there are _extra_ files in there.   i basically want to have an optional list of paths passed in to prevent these extra files from getting turned into unused assets within that submission.  they will die when the dir dies.

Reply:
ok, after a bunch of refactoring, i have now _dropped_ the `assets_added` [return](https://github.com/WildMeOrg/houston/blob/sighting_passthrough_basic/app/modules/submissions/models.py#L272) from **Submission.create_submission_from_tus()** ... this is because, even with the optional `paths=path_list` which can be passed in, in the event one (or more) of these desired files cannot be added, _the whole thing fails_ and the submission will not be made.   this means that `submission.assets` is sufficient to find _which files were added_ in both the case when `paths=` is provided.